app/models/piece.rb

```diff
@@ -1,3 +1,43 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  def obstructed?(destination_x, destination_y)
+    return true if horizontal_or_vertical_obstruction?(destination_x, destination_y)
+    return true if diagonal_obstruction?(destination_x, destination_y)
+    return true if invalid?(destination_x, destination_y)
+
+    false
```

Comment:
@mscoutermarsh does line #9 cover the invalid cases? I removed the invalid? method from the bottom of the file.